core/src/main/java/io/questdb/griffin/engine/functions/date/ToTimezoneTimestampFunctionFactory.java

```diff
@@ -0,0 +1,122 @@
+import io.questdb.griffin.engine.functions.TimestampFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.std.*;
+import io.questdb.std.datetime.TimeZoneRules;
+import io.questdb.std.datetime.microtime.TimestampFormatUtils;
+import io.questdb.std.datetime.microtime.Timestamps;
+
+import static io.questdb.std.datetime.TimeZoneRuleFactory.RESOLUTION_MICROS;
+
+public class ToTimezoneTimestampFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "to_timezone(NS)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        Function timestamp = args.getQuick(0);
+        Function timezone = args.getQuick(1);
+
+        if (timezone.isConstant()) {
+
+            CharSequence tz = timezone.getStr(null);
+            final int hi = tz.length();
+            final long l = Timestamps.parseOffset(tz, 0, hi);
+            if (l == Long.MIN_VALUE) {
+
+                try {
+                    return new OffsetTimestampFunctionFromRules(
+                            timestamp,
+                            TimestampFormatUtils.enLocale.getZoneRules(
+                                    Numbers.decodeLowInt(TimestampFormatUtils.enLocale.matchZone(tz, 0, hi)), RESOLUTION_MICROS
+                            ),
+                            1
+                    );
+                } catch (NumericException e) {
+                    Misc.free(timestamp);
+                    throw SqlException.$(argPositions.getQuick(1), "invalid timezone name");
+                }
+
+            } else {
+
+                return new OffsetTimestampFunctionFromOffset(
+                        timestamp,
+                        Numbers.decodeLowInt(l) * Timestamps.MINUTE_MICROS
+                );
+            }
+
+        } else {
+            return new ToTimezoneFunctionVar(timestamp, timezone);
+        }
+    }
+
+    private static class ToTimezoneFunctionVar extends TimestampFunction implements BinaryFunction {
+        private final Function timestamp;
+        private final Function timezone;
+
+        public ToTimezoneFunctionVar(Function timestamp, Function timezone) {
+            this.timestamp = timestamp;
+            this.timezone = timezone;
+        }
+
+        @Override
+        public Function getLeft() {
+            return timestamp;
+        }
+
+        @Override
+        public Function getRight() {
+            return timezone;
+        }
+
+        @Override
+        public long getTimestamp(Record rec) {
+            final long timestampValue = timestamp.getTimestamp(rec);
+            try {
+                return Timestamps.toTimezone(timestampValue, TimestampFormatUtils.enLocale, timezone.getStr(rec));
```

Comment:
timezone.getStr(rec) - is null value handled in toTimezone ?

Reply:
this is a value point, i'll add test